sqlglot/helper.py

```diff
@@ -421,18 +422,34 @@ def first(it: t.Iterable[T]) -> T:
     return next(i for i in it)
 
 
-def should_identify(text: str, identify: str | bool) -> bool:
+def dialects_match(this: DialectType, other: DialectType) -> bool:
+    from sqlglot.dialects import Dialect
+
```

Comment:
can't you just do this == other if they're both dialects?

Reply:
delete this entire method, overide def __eq__ on the dialect class and have it match the string

Reply:
then you just just do everywhere

dialect == "snowflake" and you're done

Reply:
`DialectType` matches strings, `Dialect`s and Type[Dialect]s too. The last one can return the dialect's name through the `__name__` attribute, which is why I also defined `__name__` in `Dialect`. So basically the reasoning was that we convert any strings to `Dialect`s and we can compare them by comparing their `__name__`s.

Reply:
but you don’t need the match method can’t you just do dialect == str

Reply:
You mean just move the logic inside Dialect's eq?

Reply:
why are you getting an actual dialect class? but yes moving logic into eq

Reply:
Sure, can try to move this into eq.

> Why are you getting an actual dialect class?

Because that's allowed in DialectType's definition right now. I guess the point is to be able to do something like:

```
from sqlglot.dialects import Snowflake

parse_one(..., read=Snowflake)
```

i.e. not have to instantiate the dialect class necessarily to use methods that expect a dialect argument.

Reply:
can you override equality for classses?

Reply:
so Snoeflake == “snowflake”

Reply:
maybe on the meta class?

Reply:
updated